app/Http/Controllers/Admin/ProductController.php

```diff
@@ -53,7 +57,16 @@ public function store(Request $request)
      */
     public function show($id)
     {
-        //
+        try {
+            $product = Product::find($id);
+            $images = Image::where('product_id', $id)->paginate(config('setting.number_paginate'), ['*'], config('setting.paginate.image'));
+            $productDetails = ProductDetail::where('product_id', $id)->paginate(config('setting.number_paginate'), ['*'], config('setting.paginate.product_detail'));
+            $comments = Comment::where('product_id', $id)->paginate(config('setting.number_paginate'), ['*'], config('setting.paginate.comment'));
```

Comment:
```suggestion
            $product = Product::with('image', 'comments', 'productDetalts')->find($id);
            $images = $product->images->paginate(config('setting.number_paginate'), ['*'], config('setting.paginate.image'));
            $productDetails = $product->productDetails->where('product_id', $id)->paginate(config('setting.number_paginate'), ['*'], config('setting.paginate.product_detail'));
            $comments = $product->comments->where('product_id', $id)->paginate(config('setting.number_paginate'), ['*'], config('setting.paginate.comment'));
```
tối ưu truy vấn nhé

Reply:
https://laracasts.com/discuss/channels/laravel/how-to-paginate-eloquent-hasmany-relation
lỗi mà chị. em tìm rồi mà không chạy.

Reply:
vậy thì `$product->images()`
em thử lại xem